modules/ROOT/pages/functions/dev_guide/internal_registry/using-an-internal-registry.adoc

```diff
@@ -0,0 +1,28 @@
+// Module included in the following assemblies
+//
+// functions/dev_guide/internal_registry/internal-registry.adoc
+
+A common use case for building and deploying Functions is to use an external container image registry, such as Quay. However, you can instead use an OpenShift Container Platform internal registry, if it has been exposed:
+
+. link:https://docs.openshift.com/container-platform/latest/registry/securing-exposing-registry.html[Expose an internal OpenShift Container Registry.]
+
+. Run the `kn func build` or `kn func deploy` function as you normally would, but specify the exposed internal registry for the `-r` parameter:
++
+[source]
+----
+$ kn func build -i my-image -r image-registry.openshift-image-registry.svc???
```

Comment:
Why the '???'

Reply:
Just wanted to keep it on the radar, I wasn't sure the command was correct.